File: backend/src/LogistiqueLesLions.Domain/Enums/NegotiationEnums.cs
namespace LogistiqueLesLions.Domain.Enums;

/// <summary>
/// Estado de una negociación. La especificación organiza la lista en tres grupos:
/// En cours · En attente · Terminées.
/// </summary>
public enum NegotiationStatus
{
    /// <summary>Hay actividad reciente entre las partes.</summary>
    EnCours = 1,
    /// <summary>Esperando respuesta de la otra parte: una oferta o un contrato pendiente.</summary>
    EnAttente = 2,
    /// <summary>Cerrada: venta verificada, rechazo o abandono.</summary>
    Terminee = 3
}

/// <summary>
/// Hitos de la cronología única de una negociación.
/// </summary>
/// <remarks>
/// La especificación muestra la operación entera como una sola línea de tiempo, desde
/// «Conversation commencée» hasta «Vente vérifiée ✓». Cada funcionalidad de la Etapa 2
/// va añadiendo sus hitos al mismo hilo.
/// </remarks>
public enum NegotiationEventType
{
    /// <summary>Conversation commencée.</summary>
    ConversationStarted = 1,
    /// <summary>Offre de X FCFA.</summary>
    OfferMade = 2,
    /// <summary>Contre-offre de X FCFA.</summary>
    CounterOffer = 3,
    /// <summary>Offre acceptée.</summary>
    OfferAccepted = 4,
    /// <summary>Offre refusée.</summary>
    OfferRejected = 5,
    /// <summary>Contrat créé.</summary>
    ContractCreated = 6,
    /// <summary>Modification demandée.</summary>
    ContractChangeRequested = 7,
    /// <summary>Contrat validé.</summary>
    ContractValidated = 8,
    /// <summary>Vente vérifiée ✓.</summary>
    SaleVerified = 9
}

File: app mobile/lib/features/negotiations/models/negotiation_enums.dart
// Etiquetas en francés de los enums de negociación (la API los envía como strings).

String negotiationStatusLabel(String? v) => switch (v) {
      'EnCours' => 'En cours',
      'EnAttente' => 'En attente',
      'Terminee' => 'Terminée',
      _ => '—',
    };

String offerStatusLabel(String? v) => switch (v) {
      'EnAttente' => 'En attente',
      'Acceptee' => 'Acceptée',
      'Refusee' => 'Refusée',
      'ContreOfferte' => 'Contre-offre',
      'Retiree' => 'Retirée',
      _ => '—',
    };

/// Texto del hito de la cronología. `byMe` ajusta la voz.
String eventLabel(String? type, {required bool byMe, String? amountLabel}) {
  final who = byMe ? 'Vous' : 'L’autre partie';
  return switch (type) {
    'ConversationStarted' => 'Conversation démarrée',
    'OfferMade' =>
      '$who ${byMe ? 'avez' : 'a'} fait une offre${amountLabel != null ? ' : $amountLabel' : ''}',
    'CounterOffer' =>
      '$who ${byMe ? 'avez' : 'a'} fait une contre-offre${amountLabel != null ? ' : $amountLabel' : ''}',
    'OfferAccepted' => '$who ${byMe ? 'avez' : 'a'} accepté l’offre',
    'OfferRejected' => '$who ${byMe ? 'avez' : 'a'} refusé l’offre',
    'ContractCreated' => 'Contrat créé',
    'ContractChangeRequested' => 'Modification du contrat demandée',
    'ContractValidated' => 'Contrat validé',
    'SaleVerified' => 'Vente vérifiée ✓',
    _ => 'Mise à jour',
  };
}
